crates/ursa-telemetry/src/lib.rs

```diff
@@ -68,14 +69,16 @@ impl TelemetryConfig {
     }
 
     pub fn init(self) -> anyhow::Result<()> {
-        let env_filter = EnvFilter::try_from_default_env().unwrap_or_else(|_| {
-            EnvFilter::new(self.log_level.unwrap_or_else(|| "INFO".to_string()))
-        });
+        let log_level = self
+            .log_level
+            .unwrap_or_else(|| env::var("RUST_LOG").unwrap_or_else(|_| "INFO".to_string()));
```

Comment:
What's the difference in this change?

Reply:
easily clonable + need to instantiate possibly multiple times (as mentioned below) It also correctly uses the provided log level, then environment, whereas before it'd override provided value with the environment unexpectedly

Reply:
Override the default env what you mean? It get the default set env otherwise it sets. Prob don't need to clone it I think the only change required in the PR is lowering log levels in ursa bin

Reply:
Well if you provide explicit log level, shouldn't be a need to load from env. For example ursa bin prefers the flag value over env and finally default. But before this change it'd get overridden regardless of the flag if env is set

Reply:
What is a flag is not provided?

Reply:
Which shouldn't be preferred and only be optional

Reply:
if not provided it falls back to environment or the default

Reply:
Okay so similar to the original code? Am I missing something?

Reply:
no, with this change flow is:
bin (flag > env > default) -> telemetry uses (optional value > env > default )

vs before:
bin (same flag>enc>default) -> telemetry (env > provided value from bin > default)

Reply:
Yes that seems accurate. Telemetry should take optional or fallback.

Reply:
Env > optional > default

Reply:
> Env > optional > default 

So env var precedent over cli flag override? we should switch the logic in ursa bin to match in that case (either provide some(level) or none and let telemetry handle env/default ). But imho i think the flag should override all not env

Reply:
I think we should remove the flag it duplicated logic

Reply:
Also if you look at it from first principals you are building telemetry to fit the need of bin while telemetry should fit the need of all. Example gateway maybe and doesn't have a flag. Food for thought

Reply:
> Also if you look at it from first principals you are building telemetry to fit the need of bin while telemetry should fit the need of all. Example gateway maybe and doesn't have a flag. Food for thought 

Ironically, gateway does have a log flag and the addition of that flag to ursa bin was inspired from there. But i think the changes fits both use cases because with a flag/explicit level, can use `with_log_level`, otherwise, not using that function would just result in env or default loaded once by telemetry

Reply:
Okay two birds with one stone. Now both don't need a flag. Also I think the gateway was before this new crate